lib/marklogic.js

```diff
@@ -369,6 +371,42 @@ function MarkLogicClient(connectionParams) {
 
   this.logger = null;
 }
+
+/**
+ * Tests if a connection is successful.
+ * @method DatabaseClient#testConnection
+ * @since 2.1
+ * @returns true if the connection is successful and false otherwise with an error message
+ */
+MarkLogicClient.prototype.testConnection = function testConnection() {
+  var connectionParams = this.connectionParams;
+  var requestOptions = mlutil.copyProperties(connectionParams);
+  requestOptions.method = 'HEAD';
+  requestOptions.path = mlutil.databaseParam(connectionParams, '/v1/ping', '?');
+  var operation = new Operation(
+      'test operation', this, requestOptions, 'empty', 'single'
```

Comment:
Should the last parameter be empty?  After experimenting with curl, it looks like the HEAD request doesn't return a body in either the success or failure cases.